src/metabase/models/notification.clj

```diff
@@ -119,19 +119,31 @@
   (validate-notification instance)
   instance)
 
+(defn- update-subscription-trigger!
+  [& args]
+  (apply (requiring-resolve 'metabase.task.notification/update-subscription-trigger!) args))
+
+(defn- delete-trigger-for-subscription!
+  [& args]
+  (apply (requiring-resolve 'metabase.task.notification/delete-trigger-for-subscription!) args))
+
 (t2/define-before-update :model/Notification
   [instance]
   (validate-notification instance)
   (when-let [unallowed-key (some #{:payload_type :payload_id :creator_id} (keys (t2/changes instance)))]
     (throw (ex-info (format "Update %s is not allowed." (name unallowed-key))
                     {:status-code 400
                      :changes     (t2/changes instance)})))
+  (when (contains? (t2/changes instance) :active)
+    (let [subscriptions (t2/select :model/NotificationSubscription
+                                   :notification_id (:id instance)
+                                   :type :notification-subscription/cron)]
+      (doseq [subscription subscriptions]
+        (if (:active instance)
+          (update-subscription-trigger! subscription)
```

Comment:
Do we always need to update the trigger? Just thinking about a case where we bulk update a large number of notifications, this could lead to a lot of work being done.